src/openfermion/utils/_trotter_qasm_conversion.py

```diff
@@ -0,0 +1,331 @@
+
+		for p in term:	# p = single pauli term
+			qid = p[0]	# Qubit index
+			pop = p[1]	# Pauli op
+
+			qids.append(qid)  # Qubit index
+			ops.append(pop)	 # Pauli operator
+
+			if pop == 'X':
+				strBas1 += "H " + str(qid) + nl	 # Hadamard
+				strBas2 += "H " + str(qid) + nl	 # Hadamard
+			elif pop == 'Y':
+				strBas1 += "Rx 1.57079632679 " + str(qid) + nl # Z --> Y
+				strBas2 += "Rx -1.57079632679 " + str(qid) + nl	 # Y --> Z
+
+		# Prep for CNOTs
+		cnotPairs = np.vstack((qids[:-1], qids[1:]))
+		cnots1 = ""
+		cnots2 = ""
+		for i in range(cnotPairs.shape[1]):
+			pair = cnotPairs[:, i]
+			cnots1 += "CNOT " + str(pair[0]) + " " + str(pair[1]) + nl
+		for i in np.arange(cnotPairs.shape[1])[::-1]:
+			pair = cnotPairs[:, i]
+			cnots2 += "CNOT " + str(pair[0]) + " " + str(pair[1]) + nl
+
+		# Exponentiating each Pauli string requires five parts
+
+		# 1. Perform basis rotations
+		ostrm.write(strBas1)
+
+		# 2. First set CNOTs
+		# Store in string, add later
+		ostrm.write(cnots1)
+
+		# 3. Rotation (Note kexp & Ntrot)
+		ostrm.write("Rz " + str(termCoeff) + " "+ str(qids[-1])+ nl)
+
+		# 4. Second set of CNOTs
+		ostrm.write(cnots2)
+
+		# 5. Rotate back to Z basis
+		ostrm.write(strBas2)
+
+
+def write_qubit_ops(op_list,ostrm):
+	'''
+	This function takes in a list of QubitOperators and sends them
+	to be exponentiated and printed to a QASM file one at a time
+
+	Params: 
+	op_list: list of QubitOperators
+	ostrm: output stream (e.g. file or StringIO object)
+
+	Returns:
+	None
+    '''
+
+	for op in op_list:
+		print_qubit_op_to_qasm(ostrm,op)
+
+
+
+
+
+
+def print_to_qasm(
+	file_path, 
+	hamiltonian, 
+	trotter_steps=1, 
+	trotter_order=1, 
+	term_ordering = [], 
+	k_exp = 1.):
+	'''
+    	This function trotterizes a Qubit hamiltonian and prints it to a QASM file
+    
+    	Params:
+	    	file_path: (string) absolute file path and file name to write
+	    			the QASM file to
+	    	hamiltonian: QubitOperator to be trotterized and printed
+	    	trotter_steps (optional): number of trotter steps (slices) for
+	    			trotterization as an integer
+	    	trotter_order (optional): order of trotterization as an integer
+	    	term_ordering (optional): list of tuples (QubitOperator terms
+	    			dictionary keys) that specifies order of terms when
+					trotterizing
+	    	k_exp (optional): exponential factor to all terms when trotterizing
+    	
+    	Returns: None
+    	
+    	Raises: IOError if invalid file path, TypeError if incorrect types
+    
+    	'''
+
+    
+	# Assert that hamiltonian is a QubitOperator
+	if not isinstance(hamiltonian, QubitOperator):
+		raise TypeError("Hamiltonian must be a QubitOperator.")
+	if not hamiltonian.terms:
+		raise TypeError("Hamiltonian must be a non-empty QubitOperator.")
+	if not isinstance(trotter_steps, int):
+		raise TypeError("trotter_steps must be an int")
+	if not isinstance(trotter_order, int):
+		raise TypeError("trotter_order must be an int")
+	if not isinstance(k_exp, float):
```

Comment:
I think insisting this is a float is too restrictive.  It will fail for integers and other types.  Boil down to the essence of what this type needs to be